src/compute_accuracy.rs

```diff
@@ -106,18 +107,21 @@ impl Default for Counts {
 struct Eval<'a> {
     embeddings: &'a Embeddings<VocabWrap, StorageViewWrap>,
     section_counts: Arc<Mutex<BTreeMap<String, Counts>>>,
+    progress: Arc<Mutex<ProgressBar>>,
 }
 
 impl<'a> Eval<'a> {
-    fn new(embeddings: &'a Embeddings<VocabWrap, StorageViewWrap>) -> Self {
+    fn new(embeddings: &'a Embeddings<VocabWrap, StorageViewWrap>, pb: ProgressBar) -> Self {
         Eval {
             embeddings,
             section_counts: Arc::new(Mutex::new(BTreeMap::new())),
+            progress: Arc::new(Mutex::new(pb)),
         }
     }
 
     /// Evaluate an analogy.
     fn eval_analogy(&self, instance: &Instance) {
+        self.progress.lock().unwrap().inc(1);
```

Comment:
Does this give a lot of lock contention? (Since individual analogies are updated pretty quickly?) Maybe it's better to increment every N updates (I think you can let the `par_iter` enumerate)?

Also, `indicatif` already uses an `RwLock` for its counter, so the `Mutex` here seems unnecessary.

Reply:
> Does this give a lot of lock contention? (Since individual analogies are updated pretty quickly?) Maybe it's better to increment every N updates (I think you can let the par_iter enumerate)?
>
> Also, indicatif already uses an RwLock for its counter, so the Mutex here seems unnecessary.

Turns out that the entire wrapping is unnecessary, the `ProgressBar' wraps `Arc<Mutex<_>>`. We can just update progress in an enumerated `.par_iter()` every `n` steps.

The updated version is just as fast as the original:

~~~
target/release/finalfusion compute-accuracy   --threads 14  2907,76s user 1,37s system 1384% cpu 3:30,10 total
~~~

Reply:
Good! Then the locks are cheap compared to the matrix-vector multiplication ;).

Reply:
BTW: nice feature!